ArticleTemplates/assets/js/bootstraps/gallery.js

```diff
@@ -1,120 +1,114 @@
+
+        el.dataset.pinchCentreX = pinchCentreX;
+        el.dataset.pinchCentreY = pinchCentreY;
+    }
 
-function moveImagesToRows(rows) {
-    let i;
-    let j;
-    let offset;
+    function calcPinchCentre(elBounds, pinchCentre, axis, dimension) {
+        var elStart = elBounds[axis];
+        var elEnd = elBounds[dimension];
+        var pinch = pinchCentre[axis];
 
-    for (i = 0; i < rows.length; i++) {
-        if (i === rows.length - 1) {
-            while (rows[i].length < rows[i - 1].length) {
-                rows[i].push(rows[i - 1].pop());
-            }
+        var pinchRel = (pinch-elStart)/(elEnd);
+        if (pinchRel > 0.75) {
+            return 100
+        } else if (pinchRel < 0.25) {
+            return 0
         } else {
-            offset = (galleryProps.imagesPerRow - 1) - rows[i].length;
-
-            if (offset > 0) {
-                for (j = 0; j < offset; j++) {
-                    if (rows[i + 1]) {
-                        rows[i].push(rows[i + 1].shift());
-                    }
-                }
-            }
+            return Math.round(pinchRel*100);
         }
     }
-}
 
-function addFlexClass() {
-    let i;
-    let j;
-
-    // for each row
-    for (i = 0; i < galleryRows.length; i++) {
-        // for each image in row
-        for (j = 0; j < galleryRows[i].length; j++) {
-            // remove flex class
-            removeFlexClass(galleryRows[i][j]);
-            // add new flex class
-            galleryRows[i][j].classList.add(`gallery__image-container--flex${galleryRows[i].length}`);
+    function bounceToInitialPosition(el) {
+        var currentScale = (el.getBoundingClientRect().width/el.offsetWidth);
+        var newScale = currentScale + (1-currentScale)/5;
+        if (newScale > 0.99 && newScale < 1.01) {
+            lockArticleSwipe(false);
+            el.style.transform = 'scale(1)';
+            el.classList.remove('touch-gallery__images--pinch');
+        } else {
+            el.style.transform = 'scale('+newScale+')';
+            requestAnimationFrame(function() {
+                bounceToInitialPosition(el);
+            });
         }
     }
 }
 
-function removeFlexClass(imageElem) {
-    let i;
+function lazyLoadImages() {
+    const options = {
+        rootMargin: '2000px',
+        threshold: 0.01
+      };
 
-    // if imageElem has class name with "flex" in it then remove class
-    for (i = 0; i < imageElem.classList.length; i++) {
-        if (imageElem.classList[i].includes('flex')) {
-            imageElem.classList.remove(imageElem.classList[i]);
-        }
-    }
+      const handleIntersection = (entries) => {
+        entries.forEach(entry => {
+          if (entry.intersectionRatio > 0) {
+            entry.target.style.backgroundImage = `url(${entry.target.dataset.src})`;
+          }
+        })
+      }
+
+      const observer = new IntersectionObserver(handleIntersection, options);
```

Comment:
I'm not sure about the minimum versions of iOS/Android that are supported but you may want to check whether `IntersectionObserver` works as expected on them.

Reply:
OK forget this ☝️, I've just seen you're using a polyfill.